From f6852e27575e095269a47f434d318b5fc8822e0a Mon Sep 17 00:00:00 2001
From: Markus Armbruster <armbru@redhat.com>
Date: Sat, 2 Nov 2013 14:10:18 +0100
Subject: [PATCH 48/81] rhel: Drop "cfi.pflash01" and "isa-ide" device

RH-Author: Markus Armbruster <armbru@redhat.com>
Message-id: <1383401418-4444-3-git-send-email-armbru@redhat.com>
Patchwork-id: 55269
O-Subject: [PATCH 7.0 qemu-kvm 2/2] rhel: Drop "cfi.pflash01" and "isa-ide" device
Bugzilla: 1001076
RH-Acked-by: Paolo Bonzini <pbonzini@redhat.com>
RH-Acked-by: Marcel Apfelbaum <marcel.a@redhat.com>
RH-Acked-by: Alex Williamson <alex.williamson@redhat.com>

From: Markus Armbruster <armbru@redhat.com>

We can drop a device from the build only when it isn't referenced from
code we build.  References come in two forms: extern symbols, and qdev
driver names.

Unresolved extern symbols pose no real danger, because they fail
safely at link time.  Note for the curious: the optimizer gets rid of
the reference to isa_ide_init() in hw/i386/pc_piix.c.

Unresolved driver names get detected only at runtime, so we better be
extra careful.  There is just one occurence of either driver name
outside the code dropped by this patch: "isa-ide" in hw/isa/pc87312.c.
Compiled only when CONFIG_PC87312 is enabled.  It isn't in our build.

"cfi.pflash01" is already no-user.  Drop it anyway, because it's easy
and safe.

Signed-off-by: Markus Armbruster <armbru@redhat.com>
---
 default-configs/x86_64-softmmu.mak | 2 --
 1 file changed, 2 deletions(-)

Signed-off-by: Miroslav Rezanina <mrezanin@redhat.com>
---
 default-configs/x86_64-softmmu.mak |    2 --
 1 files changed, 0 insertions(+), 2 deletions(-)

diff --git a/default-configs/x86_64-softmmu.mak b/default-configs/x86_64-softmmu.mak
index 05cc2a3..70d8bd9 100644
--- a/default-configs/x86_64-softmmu.mak
+++ b/default-configs/x86_64-softmmu.mak
@@ -17,11 +17,9 @@ CONFIG_FDC=y
 CONFIG_ACPI=y
 CONFIG_APM=y
 CONFIG_I8257=y
-CONFIG_IDE_ISA=y
 CONFIG_IDE_PIIX=y
 CONFIG_PIIX_PCI=y
 CONFIG_I8259=y
-CONFIG_PFLASH_CFI01=y
 CONFIG_TPM_TIS=$(CONFIG_TPM)
 CONFIG_PCI_HOTPLUG=y
 CONFIG_MC146818RTC=y
-- 
1.7.1

